# DATA CLEANING

## IMPORT

In [633]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## DEFINE

## DATA OVERVIEWE

In [634]:
data = pd.read_csv('data/domofond.csv')

In [635]:
data.head(10)

page              flat_type  \
0  https://www.domofond.ru/4-komnatnaya-kvartira-...  <span>Квартира</span>   
1  https://www.domofond.ru/2-komnatnaya-kvartira-...  <span>Квартира</span>   
2  https://www.domofond.ru/1-komnatnaya-kvartira-...  <span>Квартира</span>   
3  https://www.domofond.ru/3-komnatnaya-kvartira-...  <span>Квартира</span>   
4  https://www.domofond.ru/1-komnatnaya-kvartira-...  <span>Квартира</span>   
5  https://www.domofond.ru/1-komnatnaya-kvartira-...  <span>Квартира</span>   
6  https://www.domofond.ru/2-komnatnaya-kvartira-...  <span>Квартира</span>   
7  https://www.domofond.ru/kvartira-na-prodazhu-s...  <span>Квартира</span>   
8  https://www.domofond.ru/1-komnatnaya-kvartira-...  <span>Квартира</span>   
9  https://www.domofond.ru/kvartira-na-prodazhu-s...  <span>Квартира</span>   

                object_type                rooms              floors  \
0  <span>Новостройка</span>       <span>4</span>    <span>6/9</span>   
1  <span>Новостройка</span>       <span>2</span>  <span>11/12</span>   
2  <span>Новостройка</span>       <span>1</span>  <span>18/20</span>   
3    <span>Вторичная</span>       <span>3</span>    <span>1/5</span>   
4    <span>Вторичная</span>       <span>1</span>  <span>10/14</span>   
5    <span>Вторичная</span>       <span>1</span>   <span>6/10</span>   
6    <span>Вторичная</span>       <span>2</span>    <span>4/5</span>   
7    <span>Вторичная</span>  <span>Студия</span>   <span>9/16</span>   
8    <span>Вторичная</span>       <span>1</span>   <span>7/12</span>   
9  <span>Новостройка</span>  <span>Студия</span>   <span>9/11</span>   

                 square        kitchen_square           live_square  \
0   <span>157 м²</span>  <span>37.4 м²</span>  <span>73.7 м²</span>   
1  <span>57.6 м²</span>         <span></span>  <span>40.6 м²</span>   
2  <span>48.8 м²</span>         <span></span>         <span></span>   
3  <span>62.2 м²</span>     <span>8 м²</span>  <span>43.5 м²</span>   
4  <span>50.5 м²</span>  <span>15.2 м²</span>  <span>19.1 м²</span>   
5    <span>50 м²</span>  <span>23.1 м²</span>   <span>9.6 м²</span>   
6    <span>81 м²</span>  <span>14.1 м²</span>  <span>52.3 м²</span>   
7  <span>17.8 м²</span>         <span></span>  <span>13.6 м²</span>   
8    <span>40 м²</span>    <span>16 м²</span>  <span>12.4 м²</span>   
9  <span>21.6 м²</span>         <span></span>  <span>15.5 м²</span>   

                       price             price_per_m        build_matireal  \
0  <span>68 923 000 ₽</span>  <span>439 000 ₽</span>  <span>Монолит</span>   
1  <span>12 729 600 ₽</span>  <span>221 000 ₽</span>  <span>Монолит</span>   
2   <span>8 600 012 ₽</span>  <span>176 230 ₽</span>  <span>Монолит</span>   
3   <span>4 700 000 ₽</span>   <span>75 563 ₽</span>   <span>Панель</span>   
4  <span>12 500 000 ₽</span>  <span>247 525 ₽</span>  <span>Монолит</span>   
5  <span>14 100 000 ₽</span>  <span>282 000 ₽</span>  <span>Монолит</span>   
6   <span>9 900 000 ₽</span>  <span>122 222 ₽</span>   <span>Кирпич</span>   
7   <span>2 740 000 ₽</span>  <span>153 933 ₽</span>   <span>Кирпич</span>   
8   <span>9 500 000 ₽</span>  <span>237 500 ₽</span>  <span>Монолит</span>   
9   <span>4 550 000 ₽</span>  <span>210 648 ₽</span>  <span>Монолит</span>   

               public_date            update_date               catalog_id  \
0    <span>1/8/2019</span>  <span>9/1/2021</span>  <span>1671457770</span>   
1   <span>12/9/2020</span>  <span>9/1/2021</span>  <span>2857061520</span>   
2    <span>1/2/2020</span>  <span>9/1/2021</span>  <span>2269289168</span>   
3   <span>10/9/2020</span>  <span>9/1/2021</span>  <span>2852613956</span>   
4   <span>5/11/2020</span>  <span>9/1/2021</span>  <span>2983774635</span>   
5  <span>10/11/2020</span>  <span>9/1/2021</span>  <span>2995697558</span>   
6  <span>10/11/2020</span>  <span>9/1/2021</span>  <span>2995696729</span>   
7   <span>11/9/2020</span>  <span>9/1/2021</span>  <span>2855474979</span>   
8    <span>9/1/2021</span>  <span>9/1/2021

In [636]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43732 entries, 0 to 43731
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   page             43732 non-null  object
 1   flat_type        43732 non-null  object
 2   object_type      43732 non-null  object
 3   rooms            43732 non-null  object
 4   floors           43732 non-null  object
 5   square           43732 non-null  object
 6   kitchen_square   43732 non-null  object
 7   live_square      43732 non-null  object
 8   price            43732 non-null  object
 9   price_per_m      43732 non-null  object
 10  build_matireal   43732 non-null  object
 11  public_date      43732 non-null  object
 12  update_date      43732 non-null  object
 13  catalog_id       43732 non-null  object
 14  district_rating  43732 non-null  object
 15  underground      41424 non-null  object
dtypes: object(16)
memory usage: 5.3+ MB


Удаляем html теги `<span>` и `</span>` из таблицы.

In [637]:
for column in data.columns[1:-2]:
    data[column] = data[column].apply(lambda x: x.split('<span>')[1].split('</span>')[0])

### page

В этой колонке находятся ссылки на объявления. Пока оставлю, могу пригодится при обработки остальных колонок. Удалю в конце.

### flat_type

In [638]:
data.flat_type.value_counts()

Квартира    43732
Name: flat_type, dtype: int64

Перенесем информацию из колонки **rooms** в эту колонку, для тех записей у которых в колонке **rooms** указана **студия**.

*UPDATE*
После изучения колонки **rooms** добавил в условие **Своб. планировка**

In [639]:
data.flat_type = data.rooms.apply(lambda x: 'Студия' if x=='Студия' else \
                                  ('Своб. планировка' if x=='Своб. планировка' else 'Квартира'))

In [640]:
data.flat_type.value_counts()

Квартира            35417
Студия               8252
Своб. планировка       63
Name: flat_type, dtype: int64

### object_type

In [641]:
data.object_type.value_counts()

Новостройка    29318
Вторичная      14414
Name: object_type, dtype: int64

### rooms

In [642]:
data.rooms.value_counts()

1                   12697
2                   11851
3                    8461
Студия               8252
4                    1717
5                     547
7                      71
Своб. планировка       63
8                      33
&gt; 9                 22
9                      18
Name: rooms, dtype: int64

In [643]:
data['page'][data['rooms']=='&gt; 9'].iloc[0]

'https://www.domofond.ru/kvartira-na-prodazhu-sankt_peterburg-2924987247'

Судя по объвялению это продается целый этаж (возможно бывшое общежитие или отель).

In [644]:
data['page'][data['rooms']=='Своб. планировка'].iloc[0]

'https://www.domofond.ru/kvartira-na-prodazhu-sankt_peterburg-3063799222'

Добавил условие в обработку колонки **flat_type**

In [645]:
data.rooms = data.rooms.apply(lambda x: '9' if x=='&gt; 9' else x)

In [646]:
data.rooms.value_counts()

1                   12697
2                   11851
3                    8461
Студия               8252
4                    1717
5                     547
7                      71
Своб. планировка       63
9                      40
8                      33
Name: rooms, dtype: int64

Изменим **Студия** на 1 комнату.

In [647]:
data.rooms = data.rooms.apply(lambda x: '1' if x=='Студия' else x)

In [648]:
data.rooms.value_counts()

1                   20949
2                   11851
3                    8461
4                    1717
5                     547
7                      71
Своб. планировка       63
9                      40
8                      33
Name: rooms, dtype: int64

### floors

In [649]:
data.floors.value_counts()

2/5      900
1/5      726
5/5      719
3/5      638
4/5      520
        ... 
29/35      1
25/35      1
33/35      1
16/30      1
24/32      1
Name: floors, Length: 468, dtype: int64

Добавим новый признак **total_floors**. Общее количество этажей в доме.

In [650]:
data['total_floors'] = '0'

In [651]:
data['total_floors'] = data.floors.apply(lambda x: x.split('/')[1])

Убирем из колонки **floors** общее количество этажей в доме.

In [652]:
data.floors = data.floors.apply(lambda x: x.split('/')[0])

In [653]:
data.floors.value_counts()

2     5362
3     4131
5     3820
4     3682
1     3105
7     2626
6     2625
8     2382
9     1998
10    1980
11    1709
12    1579
13    1287
14    1163
15     975
16     857
17     822
18     719
19     605
20     575
22     413
21     388
23     369
24     330
25     148
26      34
27      28
28      12
29       5
33       2
30       1
Name: floors, dtype: int64

### square

In [654]:
data.square.value_counts()

38 м²       430
26 м²       423
22.1 м²     418
40 м²       409
35 м²       405
           ... 
200.9 м²      1
148.8 м²      1
11.9 м²       1
197.2 м²      1
249.5 м²      1
Name: square, Length: 1910, dtype: int64

Убирем квадратные метры и изменим тип данных на `float`.

In [655]:
data.square = data.square.apply(lambda x: float(x.split(' ')[0]))

### kitchen_square

In [656]:
data.kitchen_square.value_counts()

           13755
10 м²       1116
5 м²        1022
12 м²        745
11 м²        719
           ...  
49.1 м²        1
72.6 м²        1
57.7 м²        1
67 м²          1
3.7 м²         1
Name: kitchen_square, Length: 547, dtype: int64

Сначала нужно разобраться с пропусками. Данные могут отсутствовать потому, что это студия или свободная планировка. Для студий можно поставить значение **0**. Для свободной планировки так делать нежелательно. Также данные могут отсутствовать просто потому, что продавец их не указал.

In [657]:
data.kitchen_square.value_counts()

           13755
10 м²       1116
5 м²        1022
12 м²        745
11 м²        719
           ...  
49.1 м²        1
72.6 м²        1
57.7 м²        1
67 м²          1
3.7 м²         1
Name: kitchen_square, Length: 547, dtype: int64

In [658]:
# Поставим для студий значение 0.
def add_missing_square(df):
    # df - передаем пару колонок (например data[['flat_type','kitchen_square'])
    
    # Создаем пустой список
    new_row = []
    
    # Проходим с помощью цикла по всем значениям из колонок flat_type и kitchen_square.
    # Если значение в колонке flat_type - Студия, то заносим в новый список значение 0.
    # Если любое другое значение, то заносим в новый список прежнее значение из колонки kitchen_square
    for row in df.values:
        if row[0] == 'Студия':
            new_row.append('0 м²') # оставил м², чтобы было проще потом привести данные к одному типу
        else:
            new_row.append(row[1])
    
    # Преобразуем список в NumPy array
    new_row = np.array(new_row)
    
    return new_row

In [659]:
data.kitchen_square = add_missing_square(data[['flat_type','kitchen_square']])

Для оставшихся квартир и свободных планировок поставим значение `None`.

In [660]:
data.kitchen_square = data.kitchen_square.apply(lambda x: None if x=='' else x)

In [661]:
data.kitchen_square = data.kitchen_square.apply(lambda x: None if x==None else float(x[:-3])) 

### live_square

In [662]:
data.live_square.value_counts()

            10735
18 м²         632
17 м²         585
15 м²         557
16 м²         523
            ...  
120.7 м²        1
168.2 м²        1
167.4 м²        1
128.9 м²        1
108.3 м²        1
Name: live_square, Length: 1214, dtype: int64

Делаем все тоже самое, что делали с колонкой **kitchen_square**

In [663]:
data.live_square = add_missing_square(data[['flat_type','live_square']])

In [664]:
data.live_square = data.live_square.apply(lambda x: None if x=='' else x)

In [665]:
data.live_square = data.live_square.apply(lambda x: None if x==None else float(x[:-3])) 

### price

In [666]:
data.price.value_counts()

6 500 000 ₽      215
5 500 000 ₽      209
7 500 000 ₽      206
4 500 000 ₽      166
5 800 000 ₽      162
                ... 
5 809 639 ₽        1
7 435 600 ₽        1
123 081 050 ₽      1
5 305 328 ₽        1
49 474 265 ₽       1
Name: price, Length: 17622, dtype: int64

Избавимся от символа **₽** и приведем данные к числовому типу.

In [667]:
data.price = data.price.apply(lambda x: float(x.replace(' ','')[:-1]))

### price_per_m

В данном проекте **price** наш таргет. Признак **price_per_m** коррелирует с коэффициентом 1. Предсказывать оба значения не вижу смысла, поэтому просто удалю признак **price_per_m**.

In [668]:
data.drop(['price_per_m'], axis=1, inplace=True)

### build_matireal

In [669]:
data.build_matireal.value_counts()

Монолит    23736
Кирпич     16348
Панель      3410
Блоки        221
Дерево        17
Name: build_matireal, dtype: int64

Все хорошо, оставляем как есть.

### public_date, update_date

Рассмотрю оба признака сразу.

**public_date**	- дата размещения объявления.

**update_date** - дата последнего обновления объявления.

Заменю два признака одним. Рассчитаю сколько дней висит объявление.

In [670]:
data.public_date = pd.to_datetime(data.public_date, dayfirst=True)
data.update_date = pd.to_datetime(data.update_date, dayfirst=True)

In [671]:
data.public_date

0       2019-08-01
1       2020-09-12
2       2020-02-01
3       2020-09-10
4       2020-11-05
           ...    
43727   2020-12-12
43728   2020-12-11
43729   2020-11-11
43730   2020-12-12
43731   2020-10-06
Name: public_date, Length: 43732, dtype: datetime64[ns]

In [672]:
data['days_active'] = data['update_date'] - data['public_date']

Удаляем оба признака.

In [673]:
data.drop(['public_date', 'update_date'], axis=1, inplace=True)

### build_matireal

In [674]:
data.build_matireal.value_counts()

Монолит    23736
Кирпич     16348
Панель      3410
Блоки        221
Дерево        17
Name: build_matireal, dtype: int64

Все хорошо, оставляю как есть.

### catalog_id

Уникальный номер объявления. Пока оставлю. Поэксперементирую с этим признаком при построении модели.

### district_rating

In [675]:
data.district_rating.value_counts()

3,7    16928
3,8     6216
3,9     4894
4,0     4088
3,2     2825
3,6     2652
3,5     2384
3,4     1406
3,3     1202
4,1     1137
Name: district_rating, dtype: int64

Значений не так много. Оставлю признак категориальным.

### underground

In [676]:
data.underground.value_counts()

Пролетарская 2.5 км,Пролетарская 2.5 км                        834
Звездная 2.1 км,Звездная 2.1 км                                822
Парнас 699 м,Парнас 699 м                                      747
Академическая 3.3 км,Академическая 3.3 км                      673
Комендантский проспект 4.0 км,Комендантский проспект 4.0 км    661
                                                              ... 
Международная 681 м,Международная 681 м                          1
Ладожская 109 м,Ладожская 109 м                                  1
Лиговский проспект 308 м,Лиговский проспект 308 м                1
Удельная 546 м,Удельная 546 м                                    1
Владимирская 412 м,Владимирская 412 м                            1
Name: underground, Length: 5631, dtype: int64

Сначала избавлю от повторов в строке.

In [677]:
data.underground.iloc[0].split(',')[0]

'Чернышевская 605 м'

In [678]:
data.underground = data.underground.apply(lambda x: x.split(',')[0] if type(x)==str else x)

In [679]:
data.underground

0                   Чернышевская 605 м
1                      Звездная 1.0 км
2        Комендантский проспект 3.6 км
3                       Обухово 1.0 км
4                Площадь Ленина 2.8 км
                     ...              
43727                              NaN
43728            Площадь Ленина 3.6 км
43729         Лиговский проспект 877 м
43730           Сосновая Поляна 4.8 км
43731          Московские ворота 712 м
Name: underground, Length: 43732, dtype: object

Признак содержит информацию о ближайшей станции метро или ж/д станции, если квартира находится в лен. области.

Добавлю отдельным признаком ближайшую станцию метро.

In [680]:
data.underground.isna().value_counts()

False    41424
True      2308
Name: underground, dtype: int64

**2308** пропусков это порядка **5%** данных (**5.5%** если быть точным). Это достаточно много, чтобы просто удалить эти данные. С другой стороны из этого признака я могу узнать район, в котором находится квартира, что является очень важным признаком, влияющим на таргет (цена квартиры). Две квартиры в разных районах, но с одинаковыми другими параметрами будут стоить по разному. Поэтому, я считаю, что данные с пропусками в этом признаке не несут пользы и мог только навредить модели.

In [681]:
data.dropna(axis=0, subset=['underground'],inplace=True)

In [682]:
def add_metro_station(column):
    
    if len(column.split()) > 4:
        return ' '.join(column.split()[:3])
    elif len(column.split()) == 4:
        return ' '.join(column.split()[:2])
    elif len(column.split()) == 3:
        return column.split()[0]
    else:
        return column
    

In [683]:
data['metro_station'] = 0
data['metro_station'] = data.underground.apply(add_metro_station)

In [684]:
data.metro_station.unique()

array(['Чернышевская', 'Звездная', 'Комендантский проспект', 'Обухово',
       'Площадь Ленина', 'Площадь А. Невского', 'Лесная', 'Купчино',
       'Московские ворота', 'Старая деревня', 'Удельная',
       'Крестовский остров', 'Балтийская', 'Академическая',
       'Елизаровская', 'Московская', 'Сергиево (Володарская)', 'Озерки',
       'Сосновая Поляна', 'Проспект Просвещения', 'Парк Победы',
       'Василеостровская', 'Приморская', 'Проспект Ветеранов',
       'Пионерская', 'Гражданский проспект', 'Бухарестская',
       'Фрунзенская', 'Рыбацкое', 'Ладожская', 'Площадь Мужества',
       'Площадь Восстания', 'Пролетарская', 'Волковская', 'Черная речка',
       'Выборгская', 'Обводный канал', 'Парнас', 'Нарвская', 'Беговая',
       'Невский проспект', 'Улица Дыбенко', 'Проспект Большевиков',
       'Девяткино', 'Лиговский проспект', 'Гостиный двор',
       'Адмиралтейская', 'Горелово', 'Электросила', 'Звенигородская',
       'Петроградская', 'Новочеркасская', 'Шувалово', 'Сенная площадь

На основе станции метро добавлю район.

In [685]:
def add_district(column):

    districts = {
        'Адмиралтейский': ['Балтийская', 'Фрунзенская', 'Звенигородская', 'Сенная площадь',
                           'Садовая', 'Пушкинская', 'Технологический ин-т II', 'Технологический ин-т I',
                           'Спасская'],
        'Ваcилеостровский': ['Василеостровская', 'Приморская'],
        'Выборгский': ['Лесная', 'Удельная', 'Озерки', 'Проспект Просвещения', 'Выборгская',
                       'Парнас', 'Шувалово'],
        'Калининский': ['Площадь Ленина', 'Академическая', 'Гражданский проспект', 'Площадь Мужества',
                        'Ручьи', 'Политехническая', 'Пискарёвка'],
        'Кировский': ['Проспект Ветеранов', 'Нарвская', 'Ленинский проспект', 'Автово', 'Кировский завод'],
        'Колпинский': [],
        'Красногвардейский': ['Ладожская', 'Новочеркасская'],
        'Красносельский': ['Сергиево (Володарская)', 'Сосновая Поляна', 'Горелово', 'Лигово'],
        'Кронштадтский': [],
        'Курортный': [],
        'Московский': ['Звездная', 'Купчино', 'Московские ворота', 'Московская', 'Парк Победы',
                       'Электросила'],
        'Невский': ['Обухово', 'Елизаровская', 'Рыбацкое', 'Пролетарская', 'Улица Дыбенко',
                    'Проспект Большевиков', 'Ломоносовская'],
        'Петроградский': ['Крестовский остров', 'Петроградская', 'Горьковская', 'Чкаловская',
                          'Спортивная', 'Новокрестовская'],
        'Петродворцовый': [],
        'Приморский': ['Комендантский проспект', 'Старая деревня', 'Пионерская', 'Черная речка',
                       'Беговая', 'Ольгино'],
        'Пушкинский': [],
        'Фрунзенский': ['Бухарестская', 'Волковская', 'Обводный канал', 'Международная'],
        'Центральный': ['Чернышевская', 'Площадь А. Невского', 'Площадь Восстания', 'Невский проспект',
                        'Лиговский проспект', 'Гостиный двор', 'Адмиралтейская', 'Достоевская', 'Маяковская',
                        'Владимирская'],
        'Всеволожский': ['Девяткино', 'Шоссейная']
    }

    for key, value in districts.items():
        if column in value:
            return key

In [686]:
data['district'] = 0
data['district'] = data.metro_station.apply(add_district)

Убиру станции из признака `underground` и переведу километры в метры.

In [687]:
data.underground = data.underground.apply(lambda x: float(x.split()[-2]) if type(x) == str and x.split()[-1] == 'м' else
                                                   (float(x.split()[-2])*1000 if type(x) == str and x.split()[-1] == 'км' else x))

In [688]:
data.underground = data.underground.astype('float64')

### total_floors

In [689]:
data.total_floors.value_counts()

24    3183
13    2975
12    2796
25    2759
9     2639
5     2636
14    2277
18    2273
20    2027
8     1782
15    1732
16    1492
10    1455
17    1347
23    1271
19    1207
6     1147
11    1125
4     1094
22    1047
7      764
21     663
27     544
26     446
3      267
29     244
2       98
28      89
35      12
32      11
30      10
1        4
36       3
34       2
33       1
37       1
55       1
Name: total_floors, dtype: int64

Все хорошо, оставляем как есть.

### days_active

In [690]:
data.days_active.value_counts()

0 days       6259
30 days      1778
1 days        729
60 days       634
31 days       634
             ... 
887 days        1
1224 days       1
1469 days       1
1439 days       1
1001 days       1
Name: days_active, Length: 1170, dtype: int64

Избавимся от days.

In [691]:
data.days_active = data.days_active.apply(lambda x: float(str(x).split()[0]))

## CLEARED DATA

In [692]:
data

page flat_type  \
0      https://www.domofond.ru/4-komnatnaya-kvartira-...  Квартира   
1      https://www.domofond.ru/2-komnatnaya-kvartira-...  Квартира   
2      https://www.domofond.ru/1-komnatnaya-kvartira-...  Квартира   
3      https://www.domofond.ru/3-komnatnaya-kvartira-...  Квартира   
4      https://www.domofond.ru/1-komnatnaya-kvartira-...  Квартира   
...                                                  ...       ...   
43726  https://www.domofond.ru/3-komnatnaya-kvartira-...  Квартира   
43728  https://www.domofond.ru/3-komnatnaya-kvartira-...  Квартира   
43729  https://www.domofond.ru/1-komnatnaya-kvartira-...  Квартира   
43730  https://www.domofond.ru/2-komnatnaya-kvartira-...  Квартира   
43731  https://www.domofond.ru/2-komnatnaya-kvartira-...  Квартира   

       object_type rooms floors  square  kitchen_square  live_square  \
0      Новостройка     4      6   157.0            37.4         73.7   
1      Новостройка     2     11    57.6             NaN         40.6   
2      Новостройка     1     18    48.8             NaN          NaN   
3        Вторичная     3      1    62.2             8.0         43.5   
4        Вторичная     1     10    50.5            15.2         19.1   
...            ...   ...    ...     ...             ...          ...   
43726    Вторичная     3      1    81.0             9.0         57.0   
43728    Вторичная     3     14    87.5            15.2         46.4   
43729    Вторичная     1      5    43.6            10.0         18.5   
43730    Вторичная     2     18    66.0            11.0         35.0   
43731  Новостройка     2      2    61.7            10.3         32.5   

            price build_matireal  catalog_id district_rating  underground  \
0      68923000.0        Монолит  1671457770             3,7        605.0   
1      12729600.0        Монолит  2857061520             4,0       1000.0   
2       8600012.0        Монолит  2269289168             3,9       3600.0   
3       4700000.0         Панель  2852613956             3,6       1000.0   
4      12500000.0        Монолит  2983774635             3,8       2800.0   
...           ...            ...         ...             ...          ...   
43726   9750000.0         Кирпич  3070113291             3,9        256.0   
43728  13505000.0        Монолит  3068198766             3,7       3600.0   
43729  10990000.0        Монолит  2998569324             3,7        877.0   
43730   7500000.0        Монолит  3070138034             3,8       4800.0   
43731  10805225.0        Монолит  2910718912             4,0        712.0   

      total_floors  days_active           metro_station        district  
0                9        527.0            Чернышевская     Центральный  
1               12        119.0                Звездная      Московский  
2               20        343.0  Комендантский проспект      Приморский  
3                5        121.0                 Обухово         Невский  
4               14         65.0          Площадь Ленина     Калининский  
...            ...          ...                     ...             ...  
43726            5         22.0                Удельная      Выборгский  
43728           17          0.0          Площадь Ленина     Калининский  
43729           12         54.0      Лиговский проспект     Центральный  
43730           19          0.0         Сосновая Поляна  Красносельский  
43731            8         57.0       Московские ворота      Московский  

[41424 rows x 17 columns]

In [693]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 41424 entries, 0 to 43731
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   page             41424 non-null  object 
 1   flat_type        41424 non-null  object 
 2   object_type      41424 non-null  object 
 3   rooms            41424 non-null  object 
 4   floors           41424 non-null  object 
 5   square           41424 non-null  float64
 6   kitchen_square   34778 non-null  float64
 7   live_square      35060 non-null  float64
 8   price            41424 non-null  float64
 9   build_matireal   41424 non-null  object 
 10  catalog_id       41424 non-null  object 
 11  district_rating  41424 non-null  object 
 12  underground      41424 non-null  float64
 13  total_floors     41424 non-null  object 
 14  days_active      41424 non-null  float64
 15  metro_station    41424 non-null  object 
 16  district         41424 non-null  object 
dtypes: float64(6

In [694]:
data.to_csv('data/flats.csv', index=False)